### ISSUE TITLE
Anomaly on Coq 8.7, error message on 8.8, when making_ind_fun_obligation opaque or admitting it

### ISSUE THREAD

**Issue by Blaisorblade** [mattam82/Coq-Equations#120]
With equations.1.0+8.7, using `Qed.` or `Admitted.` on `val_type_ind_fun_obligation` (in defining `val_fun`) triggers an anomaly. `Qed` gives:
```
Error:
Anomaly "Uncaught exception Environ.NotEvaluableConst(1)."
Please report at http://coq.inria.fr/bugs/.
```
while `Admitted` uses `NotEvaluableConst(0)` instead.

On Coq 8.8 I get instead:
```
Error:
Obligation should be transparent but was declared opaque.
Use 'Defined' instead.
```
The bad error message is annoying, being unable to use `Admitted` is worse. I'm also not sure why I get this obligation, except that I'm not using the documented way to set up well-founded relation because I need a custom one (the size of my types must ignore embedded variables, and I need lexicographic induction on two variables).

```coq
Require Export Omega.
Require Import Coq.Relations.Relation_Operators.
Require Import Coq.Wellfounded.Lexicographic_Product.
From Equations Require Import Equations.

Inductive id : Type := Id : nat -> id.
Inductive var : Type :=
| varF : id -> var
| varB : id -> var.

Inductive ty : Set :=
| TTop : ty
| TAll : ty -> ty -> ty
| TSel : var -> ty -> ty.

Inductive tm : Set :=
| tvar : id -> tm.

Fixpoint tsize_flat(T: ty) :=
  match T with
    | TTop => 1
    | TAll T1 T2 => S (tsize_flat T1 + tsize_flat T2)
    | TSel _ U => 1 + tsize_flat U
  end.

Definition val_type_termRel := MR (lexprod lt (fun _ => lt)) (fun p => let '(T, n) := p in (existT (fun _ => nat) n (tsize_flat T))).

Ltac smaller_n := autounfold; apply left_lex; omega.

Instance WF_val_type_termRel: WellFounded val_type_termRel.
  apply measure_wf; apply wf_lexprod; intro; apply lt_wf.
Qed.

Equations val_type (Tn: ty * nat) : Prop :=
  val_type Tn by rec Tn val_type_termRel :=
    val_type (pair T (S n)) := val_type (pair T n);
    val_type (pair T O) := True.

Next Obligation. smaller_n. Qed.
Next Obligation.
  induction n; simp val_type.
Defined.
(* gives: *)
(* Error: *)
(* Anomaly "Uncaught exception Environ.NotEvaluableConst(1)." *)
(* Please report at http://coq.inria.fr/bugs/. *)
```

**Comment by mattam82** [comment]
Indeed that's annoying but will require a change in Coq to allow such obligations to be declared opaque. Indeed the proof automation should not fail here, I'll investigate.

**Comment by mattam82** [comment]
Fixed in 8.8 as well.
